client/payment-methods/index.js

```diff
@@ -121,9 +121,27 @@ const PaymentMethods = () => {
 	const availablePaymentMethodIds = useGetAvailablePaymentMethodIds();
 
 	// We filter link payment method since this will be displayed in other section (express checkout).
-	const availableMethods = availablePaymentMethodIds
-		.filter( ( id ) => 'link' !== id )
-		.map( ( methodId ) => methodsConfiguration[ methodId ] );
+	// We further split the available methods into pay later and non-pay later methods to sort them in the required order later.
+	const availableNonPayLaterMethods = availablePaymentMethodIds.filter(
+		( id ) =>
+			'link' !== id &&
```

Comment:
Can we introduce an enum for housing those constants (to utilize in this PR)?

Reply:
Discussed here https://github.com/Automattic/woocommerce-payments/pull/6398#discussion_r1213259045

Reply:
Yep, the direction looks good. I will leave a further detailed review for the team.